cli/medperf/commands/compatibility_test.py

```diff
@@ -108,36 +109,26 @@ def prepare_test(self):
             self.set_cube_uid("data_prep", benchmark.data_preparation)
             self.set_cube_uid("model", benchmark.reference_model)
             self.set_cube_uid("evaluator", benchmark.evaluator)
-            demo_dataset_url = benchmark.demo_dataset_url
-            demo_dataset_hash = benchmark.demo_dataset_hash
+            self.demo_dataset_url = benchmark.demo_dataset_url
+            self.demo_dataset_hash = benchmark.demo_dataset_hash
         else:
             self.set_cube_uid("data_prep")
             self.set_cube_uid("model")
             self.set_cube_uid("evaluator")
-            demo_dataset_url = None
-            demo_dataset_hash = None
-
-        self.benchmark = Benchmark.tmp(
-            self.data_prep,
-            self.model,
-            self.evaluator,
-            demo_dataset_url,
-            demo_dataset_hash,
-        )
-        self.benchmark_uid = self.benchmark.uid
 
     def execute_benchmark(self):
         """Runs the benchmark execution flow given the specified testing parameters
         """
+        benchmark = Benchmark.tmp(self.data_prep, self.model, self.evaluator)
         BenchmarkExecution.run(
-            self.benchmark_uid,
+            benchmark.uid,
             self.data_uid,
             self.model,
             self.comms,
             self.ui,
             run_test=True,
         )
-        return Result(self.benchmark_uid, self.dataset.uid, self.model)
+        return Result(benchmark.uid, self.dataset.generated_uid, self.model)
```

Comment:
This is because of the changes you did to how compatibility-test results are generated? If so it would be good to add a little comment here

Reply:
This is because compatibility tests can use datasets that are not registered on the server (dataset.uid = None). Before, only registered datasets are expected to be used in compatibility tests.

So now results of real executions (`medperf run`) contain dataset server uid, and results of compatibility tests (`medperf test`) contain the generated uid (those results are not expected to be uploaded to the server, right?).
I will add a comment accordingly